cpp_tests/source/test_cell_lists.cpp

```diff
@@ -1167,3 +1169,70 @@ TEST(LatticeNeighborsTest, NonPeriodic_Works2)
     neibs = lattice.find_all_neighbors(2);
     ASSERT_EQ(neibs.size(), size_t(2*3*2));
 }
+
+using pele::SimplePairwisePotential;
+using pele::lj_interaction_cut_smooth;
+using pele::periodic_distance;
+
+template<size_t NDIM>
+class LJCutPeriodicN : public SimplePairwisePotential<lj_interaction_cut_smooth, periodic_distance<NDIM> > {
+public:
+    LJCutPeriodicN(double C6, double C12, double rcut, Array<double> const boxvec)
+        : SimplePairwisePotential< lj_interaction_cut_smooth, periodic_distance<NDIM> > (
+                std::make_shared<lj_interaction_cut_smooth>(C6, C12, rcut),
+                std::make_shared<periodic_distance<NDIM> >(boxvec)
+                ) 
+    {}
+};
+
+template<size_t ndim>
+void test_rectangular_cell_lists()
+{
+    double rcut = 2.5;
+    Array<double> boxvec(ndim);
+    for (size_t i = 0; i < ndim; ++i) {
+        boxvec[i] = std::pow((2 * rcut + 1), i + 1) + 0.8 * i;
+    }
+    const size_t N = 25;
+    pele::Array<double> x(ndim * N);
+    std::mt19937_64 gen(42);
+    for (size_t i = 0; i < N; ++i) {
+        x[i * ndim] = std::uniform_real_distribution<double>{0, boxvec[0]}(gen);
+        x[i * ndim + 1] = std::uniform_real_distribution<double>{0, boxvec[1]}(gen);
+    }
```

Comment:
does it make sense to add another test where the atoms are not all in the [0,L] box.  e.g. in [-L/2, L/2] or scattered over all space.